WordPress/src/main/java/org/wordpress/android/ui/stats/StatsSinglePostDetailsActivity.java

```diff
@@ -0,0 +1,711 @@
+            if (numberOfChilds > 1) {
+                PostViewsModel.Day lastChild = (PostViewsModel.Day) getChild(groupPosition, getChildrenCount(groupPosition) - 1);
+                name = StatsUtils.parseDate(firstChild.getDay(), "yyyy-MM-dd", "MMM dd")
+                        + " - " + StatsUtils.parseDate(lastChild.getDay(), "yyyy-MM-dd", "MMM dd");
+            } else {
+                name = StatsUtils.parseDate(firstChild.getDay(), "yyyy-MM-dd", "MMM dd");
+            }
+
+            if (shouldShowTheTrophyIcon) {
+                holder.setEntryText(name, getResources().getColor(R.color.calypso_orange));
+            } else {
+                holder.setEntryText(name, getResources().getColor(R.color.stats_link_text_color));
+            }
+
+            holder.networkImageView.setVisibility(View.GONE);
+
+            // totals
+            holder.totalsTextView.setText(FormatUtils.formatDecimal(total));
+
+            // expand/collapse chevron
+            holder.chevronImageView.setVisibility(numberOfChilds > 0 ? View.VISIBLE : View.GONE);
+            return convertView;
+        }
+
+        @Override
+        public boolean hasStableIds() {
+            return false;
+        }
+
+        @Override
+        public boolean isChildSelectable(int groupPosition, int childPosition) {
+            return false;
+        }
+
+    }
+
+
+    private class MonthsAndYearsListAdapter extends BaseExpandableListAdapter {
+        public final LayoutInflater inflater;
+        private final List<PostViewsModel.Year> groups;
+        private final int maxReachedValue;
+
+        public MonthsAndYearsListAdapter(Context context, List<PostViewsModel.Year> groups, int maxReachedValue) {
+            this.groups = groups;
+            this.inflater = LayoutInflater.from(context);
+            this.maxReachedValue = maxReachedValue;
+        }
+
+        @Override
+        public Object getChild(int groupPosition, int childPosition) {
+            PostViewsModel.Year currentYear = groups.get(groupPosition);
+            return currentYear.getMonths().get(childPosition);
+        }
+
+        @Override
+        public long getChildId(int groupPosition, int childPosition) {
+            return 0;
+        }
+
+        @Override
+        public View getChildView(int groupPosition, final int childPosition,
+                                 boolean isLastChild, View convertView, ViewGroup parent) {
+
+            final PostViewsModel.Month currentMonth = (PostViewsModel.Month) getChild(groupPosition, childPosition);
+
+            if (convertView == null) {
+                convertView = inflater.inflate(R.layout.stats_list_cell, parent, false);
+                // configure view holder
+                StatsViewHolder viewHolder = new StatsViewHolder(convertView);
+                convertView.setTag(viewHolder);
+            }
+
+            final StatsViewHolder holder = (StatsViewHolder) convertView.getTag();
```

Comment:
Rather than retrieve the holder that may have just been created above, this would be more optimal:

```
final StatsViewHolder holder;
if (convertView == null) {
    convertView = inflater.inflate(R.layout.stats_list_cell, parent, false);
    holder = new StatsViewHolder(convertView);
    convertView.setTag(holder);
} else {
    holder = (StatsViewHolder) convertView.getTag();
}
```